src/api/metadata/types.ts

```diff
@@ -1,17 +1,30 @@
-interface Metadata {
+type Metadata = {
   version: string
 }
 
-export interface ReferralMetadata extends Metadata {
+export type ReferralMetadata = Metadata & {
   address: string
 }
 
-export interface QuoteMetadata extends Metadata {
-  id?: string
+export type OnlyQuoteAmounts = {
   sellAmount: string
   buyAmount: string
+  slippageInBips?: never
 }
 
+export type OnlyQuoteSlippage = {
+  sellAmount?: never
+  buyAmount?: never
+  slippageInBips: string
+}
+
+export type OnlyQuoteAmountsOrSlippage = OnlyQuoteAmounts | OnlyQuoteSlippage
```

Comment:
Typescript trick to make either types required.

Set `sellAmount` & `buyAmount` OR `slippageInBips`